src/main/java/io/strimzi/test/container/KafkaVersionService.java

```diff
@@ -135,6 +138,29 @@ public static int compareVersions(String version1, String version2) {
             return components.length - otherComponents.length;
         }
 
+        /**
+         * Extracts the Kafka version from a Docker image name.
+         *
+         * <p>Expects the image name to contain "-kafka-" followed by the version number at the end.
+         * For example:
+         * <ul>
+         *   <li>"quay.io/strimzi-test-container/test-container:0.107.0-rc1-kafka-3.7.1" returns "3.7.1"</li>
+         *   <li>"quay.io/strimzi-test-container/test-container:0.105.0-kafka-3.6.0" returns "3.6.0"</li>
+         * </ul>
+         *
+         * @param imageName the Docker image name (e.g., "quay.io/...:0.107.0-rc1-kafka-3.7.1")
+         * @return the extracted Kafka version (e.g., "3.7.1")
+         * @throws IllegalArgumentException if the version cannot be extracted
+         */
+        public static String extractVersionFromImageName(final String imageName) {
+            final Matcher matcher = KAFKA_VERSION_PATTERN.matcher(imageName);
+            if (matcher.find()) {
+                return matcher.group(1); // Returns the Kafka version string like "3.9.0"
```

Comment:
This is pretty unreliable. I think we need to have better way to configure KAfka version than through the container image. If you do not want to solve it in this PR it might be fine, but we should have issues to track this.

This is pretty likel to fail when using custom images when in various ofline / disconnected environments, when you want to fix to use some specific image through digest etc.

Reply:
Yeah, I was thinking about the same with custom images. I can create an issue when I solve that also for custom images or at least log a warning when using custom images.

Reply:
I think it should work like this:
* The builders should have a `withVersion` field/method
* If not set, the default version (the latest Kafka version) should be used
* The image should be by the users always set to correspond to the version

I think that is the way it should work. (As I said, I'm happy to leave it for later and not block this PR.)

Reply:
Yeah, I think I have come up with a more clear solution. Let me know what you think